src/plugins/rgbcolor/rgbcolor.c

```diff
@@ -0,0 +1,379 @@
+					  { "mediumspringgreen", 0x00fa9aff },
+					  { "mediumturquoise", 0x48d1ccff },
+					  { "mediumvioletred", 0xc71585ff },
+					  { "midnightblue", 0x191970ff },
+					  { "mintcream", 0xf5fffaff },
+					  { "mistyrose", 0xffe4e1ff },
+					  { "moccasin", 0xffe4b5ff },
+					  { "navajowhite", 0xffdeadff },
+					  { "navy", 0x000080ff },
+					  { "oldlace", 0xfdf5e6ff },
+					  { "olive", 0x808000ff },
+					  { "olivedrab", 0x6b8e23ff },
+					  { "orange", 0xffa500ff },
+					  { "orangered", 0xff4500ff },
+					  { "orchid", 0xda70d6ff },
+					  { "palegoldenrod", 0xeee8aaff },
+					  { "palegreen", 0x98fb98ff },
+					  { "paleturquoise", 0xafeeeeff },
+					  { "palevioletred", 0xdb7093ff },
+					  { "papayawhip", 0xffefd5ff },
+					  { "peachpuff", 0xffdab9ff },
+					  { "peru", 0xcd853fff },
+					  { "pink", 0xffc0cbff },
+					  { "plum", 0xdda0ddff },
+					  { "powderblue", 0xb0e0e6ff },
+					  { "purple", 0x800080ff },
+					  { "red", 0xff0000ff },
+					  { "rosybrown", 0xbc8f8fff },
+					  { "royalblue", 0x4169e1ff },
+					  { "saddlebrown", 0x8b4513ff },
+					  { "salmon", 0xfa8072ff },
+					  { "sandybrown", 0xf4a460ff },
+					  { "seagreen", 0x2e8b57ff },
+					  { "seashell", 0xfff5eeff },
+					  { "sienna", 0xa0522dff },
+					  { "silver", 0xc0c0c0ff },
+					  { "skyblue", 0x87ceebff },
+					  { "slateblue", 0x6a5acdff },
+					  { "slategray", 0x708090ff },
+					  { "slategrey", 0x708090ff },
+					  { "snow", 0xfffafaff },
+					  { "springgreen", 0x00ff7fff },
+					  { "steelblue", 0x4682b4ff },
+					  { "tan", 0xd2b48cff },
+					  { "teal", 0x008080ff },
+					  { "thistle", 0xd8bfd8ff },
+					  { "tomato", 0xff6347ff },
+					  { "turquoise", 0x40e0d0ff },
+					  { "violet", 0xee82eeff },
+					  { "wheat", 0xf5deb3ff },
+					  { "white", 0xffffffff },
+					  { "whitesmoke", 0xf5f5f5ff },
+					  { "yellow", 0xffff00ff },
+					  { "yellowgreen", 0x9acd32ff },
+					  { NULL, 0 } };
+
+typedef enum
+{
+	COLOR_INVALID,
+	HEX_THREE,
+	HEX_FOUR,
+	HEX_SIX,
+	HEX_EIGHT,
+	NAMED_COLOR
+} ColorVariant;
+
+static const NamedColor * findNamedColor (const char * colorName)
+{
+	const NamedColor * cur = &NamedColors[0];
+	while (cur->name != NULL)
+	{
+		if (strcmp (cur->name, colorName) == 0)
+		{
+			return cur;
+		}
+		++cur;
+	}
+	return NULL;
+}
+
+static ColorVariant is_valid_key (Key * key, Key * parentKey)
+{
+	const Key * meta = keyGetMeta (key, "check/rgbcolor");
+	if (!meta) return 1;
+	const char * value = keyString (key);
+	const char * regexString = "^#([0-9a-fA-F]{3,4}|[0-9a-fA-F]{6}|[0-9a-fA-F]{8})$";
+
+	regex_t regex;
+	regmatch_t offsets;
+	int compile_failure = regcomp (&regex, regexString, REG_NOSUB | REG_EXTENDED | REG_NEWLINE);
+	if (compile_failure) return -1;
+	// regexec returns 0 on a successful match
+	int match = !(regexec (&regex, value, 0, &offsets, 0));
+	regfree (&regex);
+
+	const NamedColor * namedColor = findNamedColor (value);
+
+	if (namedColor != NULL)
```

Comment:
I moved the check before the hex-check. I assume you suggested this because it is faster than the regex check?

Reply:
> I assume you suggested this because it is faster than the regex check?

Not necessarily faster, but you always computed the regex result, even if you didn't use it.

PS. You don't need to answer all the comments, if you don't have questions and just did want I suggested a thumbs up (or no reaction at all) is fine.